adapter/src/ethereum.rs

```diff
@@ -422,6 +431,182 @@ pub fn ewt_verify(
     Ok(verified_payload)
 }
 
+mod error {
+    use primitives::adapter::{AdapterErrorKind, Error as AdapterError};
+    use primitives::ChannelId;
+    use std::fmt;
+
+    #[derive(Debug)]
+    pub enum Error {
+        Keystore(KeystoreError),
+        WalletUnlock(ethstore::Error),
+        Web3(web3::Error),
+        RelayerClient(reqwest::Error),
+        /// When the ChannelId that we get from hashing the EthereumChannel with the contract address
+        /// does not align with the provided Channel
+        InvalidChannelId {
+            expected: ChannelId,
+            actual: ChannelId,
+        },
+        ChannelInactive(ChannelId),
+        /// Signing of the message failed
+        SignMessage(EwtSigningError),
+        VerifyMessage(EwtVerifyError),
+        ContractInitialization(ethabi::Error),
+        ContractQuerying(web3::contract::Error),
+        StateRoot(StateRootError),
+    }
+
+    impl std::error::Error for Error {}
+
+    impl AdapterErrorKind for Error {}
+
+    impl fmt::Display for Error {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            use Error::*;
+
+            match self {
+                Keystore(err) => write!(f, "Keystore error - {}", err),
```

Comment:
This format isn't great for parsing error logs. We should choose an error format  that all errors should adhere to

I suggest the below:

```
Title: Description : error  
```
e.g.

```
PublicKeyRecovery : Recovering the public key from the signature : {}
```

or

```
PublicKeyRecovery :  {err}
```

Reply:
Since the errors are nested you actually get something like:
`Mar 11 12:58:16.198 ERRO sentry: Adapter specific: Keystore: "address" is invalid: Invalid argument Keystore ValidatorId, module: middleware-auth`
When the error bubbles up to the logging and starts composing the error message, it will get all those nested errors and nest them as `Error msg 1: Error msg 2: Error msg 3: Error msg 4: Error msg 5`

I did however removed the `error` part and replaced the `-` with `:`.